test/server/admin/grouped_stats_request_test.cc

```diff
@@ -66,159 +67,211 @@ class PrometheusStatsFormatterTest : public testing::Test {
-                                symbol_table_->toString(name_tag.second)));
+  std::string response(GroupedStatsRequest& request) {
+    Http::TestResponseHeaderMapImpl response_headers;
+    Http::Code code = request.start(response_headers);
+    EXPECT_EQ(Http::Code::OK, code);
+    Buffer::OwnedImpl data;
+    while (request.nextChunk(data)) {
     }
-    return result;
+    return data.toString();
   }
 
   void clearStorage() {
     pool_.clear();
-    counters_.clear();
-    gauges_.clear();
-    histograms_.clear();
-    textReadouts_.clear();
+    test_scope_ptr_->counters_.clear();
+    test_scope_ptr_->histograms_.clear();
+    test_scope_ptr_->gauges_.clear();
+    test_scope_ptr_->text_readouts_.clear();
     EXPECT_EQ(0, symbol_table_->numSymbols());
   }
 
   Stats::TestUtil::TestSymbolTable symbol_table_;
   Stats::AllocatorImpl alloc_;
   Stats::StatNamePool pool_;
-  std::vector<Stats::CounterSharedPtr> counters_;
-  std::vector<Stats::GaugeSharedPtr> gauges_;
-  std::vector<Stats::ParentHistogramSharedPtr> histograms_;
-  std::vector<Stats::TextReadoutSharedPtr> textReadouts_;
+  Stats::CustomStatNamespacesImpl custom_namespaces_;
+  testing::NiceMock<MockStore> mock_store_;
+  std::shared_ptr<TestScope> test_scope_ptr_ = std::make_shared<TestScope>("", mock_store_);
 };
 
-TEST_F(PrometheusStatsFormatterTest, MetricName) {
-  Stats::CustomStatNamespacesImpl custom_namespaces;
-  std::string raw = "vulture.eats-liver";
-  std::string expected = "envoy_vulture_eats_liver";
-  auto actual = PrometheusStatsFormatter::metricName(raw, custom_namespaces);
-  EXPECT_TRUE(actual.has_value());
-  EXPECT_EQ(expected, actual.value());
+class HistogramWrapper {
+public:
+  HistogramWrapper() : histogram_(hist_alloc()) {}
+
+  ~HistogramWrapper() { hist_free(histogram_); }
+
+  const histogram_t* getHistogram() { return histogram_; }
+
+  void setHistogramValues(const std::vector<uint64_t>& values) {
+    for (uint64_t value : values) {
+      hist_insert_intscale(histogram_, value, 0, 1);
+    }
+  }
+
+  void setHistogramValuesWithCounts(const std::vector<std::pair<uint64_t, uint64_t>>& values) {
+    for (std::pair<uint64_t, uint64_t> cv : values) {
+      hist_insert_intscale(histogram_, cv.first, 0, cv.second);
+    }
+  }
+
+private:
+  histogram_t* histogram_;
+};
+
+TEST_F(GroupedStatsRequestTest, Empty) { EXPECT_EQ(0, iterateChunks(*makeRequest(false))); }
+
+TEST_F(GroupedStatsRequestTest, OneCounter) {
+  Stats::StatNameTagVector c1Tags{{makeStatName("cluster"), makeStatName("c1")}};
+  Stats::Counter& c1 = store_.rootScope()->counterFromStatNameWithTags(
+      makeStatName("cluster.upstream.cx.total"), c1Tags);
+  c1.add(10);
+
+  EXPECT_EQ(1, iterateChunks(*makeRequest(false)));
 }
 
-TEST_F(PrometheusStatsFormatterTest, SanitizeMetricName) {
-  Stats::CustomStatNamespacesImpl custom_namespaces;
-  std::string raw = "An.artist.plays-violin@019street";
```

Comment:
this is great. I just want to leave a note for @ggreenway for senior maintainer review that these previous tests were moved to the prometheus section of render_test.cc, not removed.